staging/src/k8s.io/client-go/transport/config.go

```diff
@@ -68,7 +68,11 @@ type Config struct {
 	WrapTransport WrapperFunc
 
 	// Dial specifies the dial function for creating unencrypted TCP connections.
+	// If specified, this transport will be non-cacheable unless DialHolder is also set.
 	Dial func(ctx context.Context, network, address string) (net.Conn, error)
+	// DialHolder can be populated to make transport configs cacheable.
+	// If specified, DialHolder.Dial must be equal to Dial.
```

Comment:
the other option is to give preference to DialHolder, is how http.Transport implemented the change from `Dial` to `DialContext` in the std lib

Reply:
having parallel fields which silently override each other is pretty confusing when multiple actors start interacting with a single config... I think we should start by requiring them to be coherent to avoid accidents